mypy/build.py

```diff
@@ -1146,7 +1155,14 @@ def is_fresh(self) -> bool:
         # self.meta.dependencies when a dependency is dropped due to
         # suppression by --silent-imports.  However when a suppressed
         # dependency is added back we find out later in the process.
-        return self.meta is not None and self.dependencies == self.meta.dependencies
+        return (self.meta is not None
+                and self.dependencies == self.meta.dependencies
+                and self.child_modules == self.meta.child_modules)
+
+    def has_new_submodules(self) -> bool:
+        """Returns whether this module has any new submodules after it was
```

Comment:
I prefer having either a one-line docstring or a multi-line one that has a blank line as the second line -- the first line should count as a brief summary, brief enough to fit on one line. (I think PEP 257 describes this.)